feature/text_based.py

```diff
@@ -0,0 +1,724 @@
+    multi-level optimization framework.
+
+    The class also includes baseline implementations that selects content based on a KMeans clustering of the content
+    features (kmeans) or simple random selection (random).
+
+    Attributes
+    ----------
+    selection_size: int
+        Maximum number of content to select.
+    seed: int
+        Random seed.
+    trials: int
+        Number of trials to run for random functions to estimate average coverage.
+    verbose: bool
+        Print intermediate results if True.
+    matrix: np.ndarray
+        Coverage matrix where each row corresponds to all predefined labels for each category and the columns
+        correspond to the content.
+    features: np.ndarray
+        Numeric featurization of content.
+
+    """
+
+    def __init__(self,  selection_size: int, seed: int = 123546, trials: int = 100, verbose: int = 0):
+        """ContentSelector
+
+       Creates a content selector object.
+
+       Parameters
+       ----------
+        selection_size: int
+            Maximum number of content to select.
+        seed: int, default=123456
+            Random seed.
+        trials: int, default=100
+            Number of trials to run for random functions to estimate average coverage.
+        verbose: int, default=0
+            Print intermediate results if 1.
+
+        """
+
+        # Validate arguments
+        self._validate_args(selection_size)
+
+        # Set input arguments
+        self.selection_size = selection_size
+        self.seed = seed
+        self.trials = trials
+        self.verbose = verbose == 1
+
+
+        # Initialize class variables
+        self.matrix = None
+        self.features = None
+        self._num_rows = None
+        self._num_cols = None
+
+    def run_content_selection(self,
+                              input_df: pd.DataFrame,
+                              categories: List[int], featurization_method: TextWiser, method: str = "random",
+                              cost_metric: Optional[str] = None) -> List:
+
+        """Run content selection algorithm.
+
+        Parameters
+        ----------
+        input_df: pd.DataFrame
+            Input data frame with categories and features of content to select from.
+
+        categories: List[int]
+            List of columns in data that contains categories/labels to be covered.
+
+        featurization_method: TextWiser
+
+        method: str, default="exact"
+            Method used to perform content selection. Supported options are:
+            - "exact" maximizes content diversity and minimizes the number of content to cover all content labels.
+            - "kmeans" clusters the content into selection_size number of clusters and then selects the items closest
+              to the centroid of each of the clusters. This method does not consider the content categories/labels.
+            - "greedy" performs greedy heuristic selecting items with max unit coverage until all items are covered
```

Comment:
The kmeans line has the "This method does not consider the content labels .." and analogously, greedy can have "This method does not consider the diversity"

Reply:
Shall we remove the definitions of optimization algorithms from here? We extensively explain each of them in the class TextBased in selector.py.

Reply:
YES! I like that suggestion. There are good information here too, so please "merge" all them under selector.py

Reply:
(and here drop a comment that "see TextBased in Selector.py" or sth